docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -923,175 +951,254 @@ type PrecompileUpgrade struct {
-	})
+We can now set the following paths. `AVALANCHEGO_EXEC_PATH` points to the latest AvalancheGo binary
+we have just built. `AVALANCHEGO_PLUGIN_PATH` points to the plugins path which should have the
+Subnet-EVM binary we have just built:
 
-	// ADD YOUR PRECOMPILE HERE
-	/*
-			ginkgo.It("your precompile", ginkgo.Label("solidity-with-npx"), func() {
-			err := startSubnet("./tests/e2e/genesis/{your_precompile}.json")
-			gomega.Expect(err).Should(gomega.BeNil())
-			running := runner.IsRunnerUp(grpcEp)
-			gomega.Expect(running).Should(gomega.BeTrue())
-			runHardhatTests("./test/{YourPrecompileTest}.ts")
-			stopSubnet()
-			running = runner.IsRunnerUp(grpcEp)
-			gomega.Expect(running).Should(gomega.BeFalse())
-		})
-	*/
+```bash
+export AVALANCHEGO_EXEC_PATH="${GOPATH}/src/github.com/ava-labs/avalanchego/build/avalanchego"
+export AVALANCHEGO_PLUGIN_PATH="${GOPATH}/src/github.com/ava-labs/avalanchego/build/plugins"
 ```
 
-### Step 10: Run E2E Test
-
-We also need to modify the genesis in `./scripts/run.sh` to enable our precompile.
-Copy and paste the code snippet below and add it to the genesis located in `./scripts/run.sh`:
+The following command will "issue requests" to the server we just spun up. We can use
+avalanche-network-runner to spin up some nodes that run the latest version of Subnet-EVM:
 
-```json
-"helloWorldConfig": {
-  "blockTimestamp": 0
-},
+```bash
+  avalanche-network-runner control start \
+  --log-level debug \
+  --endpoint="0.0.0.0:8080" \
+  --number-of-nodes=5 \
+  --avalanchego-path ${AVALANCHEGO_EXEC_PATH} \
+  --plugin-dir ${AVALANCHEGO_PLUGIN_PATH} \
+  --blockchain-specs '[{"vm_name": "subnetevm", "genesis": "./tests/precompile/genesis/hello_world.json"}]'
 ```
 
-Kill any previous avalanche processes from the previous steps.
-Now we can run our script with some flags specified:
+We can look at the server terminal tab and see it booting up the local network.
+If the network startup is successful then you should see something like this:
 
 ```bash
-pkill avalanche
-
-cd $GOPATH/src/github.com/ava-labs/subnet-evm
-SKIP_NETWORK_RUNNER_START=true SKIP_NETWORK_RUNNER_SHUTDOWN=true ENABLE_SOLIDITY_TESTS=true scripts/run.sh
+[blockchain RPC for "srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"] "http://127.0.0.1:9650/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU"
+[blockchain RPC for "srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"] "http://127.0.0.1:9652/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU"
+[blockchain RPC for "srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"] "http://127.0.0.1:9654/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU"
+[blockchain RPC for "srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"] "http://127.0.0.1:9656/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU"
+[blockchain RPC for "srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"] "http://127.0.0.1:9658/ext/bc/2jDWMrF9yKK8gZfJaaaSfACKeMasiNgHmuZip5mWxUfhKaYoEU"
 ```
 
```

Comment:
Would it be useful to explain how we could use the rpc URLs to interact with the precompile. Not saying we need a remix tutorial, but maybe ideas on what they can do with this or a link to the metamask article we have published.

Reply:
Yup good idea, I'll add this back